### PR TITLE
Adding integration test for elasticsearch Metricbeat module, xpack code path

### PR DESCRIPTION
**Pull request by ycombinator**
## What does this PR do?

This PR:
- adds an integration test for the Metricbeat `elasticsearch` module when `xpack.enabled: true` is set in the module configuration, and
- does some minor cleanup and refactoring in the Metricbeat `elasticsearch` integration test code.

## Why is it important?

The code path in the `elasticsearch` module when `xpack.enabled` was set to `true` was previously not being exercised by integration tests. Moreover, it's a critical code path as it powers the Stack Monitoring feature.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~- [ ] I have made corresponding changes to the documentation~
~- [ ] I have made corresponding change to the default configuration files~
- [x] I have added tests that prove my fix is effective or that my feature works

## Related issues

- Requires #15800, as it introduces some necessary changes to the integration tests framework itself.

### CODE REVIEW

File: metricbeat/module/elasticsearch/elasticsearch_integration_test.go [hunk 1]
```diff
@@ -124,6 +122,92 @@ func TestData(t *testing.T) {
...
+				}
+
+				return
+			}
+
+			// Special case: the `shard` metricset generates as many events
+			// as there are distinct shards in Elasticsearch
+			if metricSet.Name() == "shard" {
+				numShards, err := countShards(host)
+				if !assert.NoError(t, err) {
```
Comment (mtojek) [inline metricbeat/module/elasticsearch/elasticsearch_integration_test.go]: `require.NoError()`

Comment (ycombinator) [reply metricbeat/module/elasticsearch/elasticsearch_integration_test.go]: That's awesome, TIL!

I'll updated the usages of `assert.NoError` to `require.NoError` in this PR here and make a follow up PR to fix them in other tests.

Thanks!

File: metricbeat/module/elasticsearch/elasticsearch_integration_test.go [hunk 2]
```diff
@@ -124,6 +123,78 @@ func TestData(t *testing.T) {
 	}
 }
 
+func TestXPackEnabled(t *testing.T) {
+	service := compose.EnsureUpWithTimeout(t, 300, "elasticsearch")
+	host := service.Host()
+
+	version, err := getElasticsearchVersion(host)
+	if err != nil {
```
Comment (mtojek) [inline metricbeat/module/elasticsearch/elasticsearch_integration_test.go]: I think you apply `require` also here.

File: metricbeat/module/elasticsearch/elasticsearch_integration_test.go (hunk 1 above)
Comment (ycombinator) [reply metricbeat/module/elasticsearch/elasticsearch_integration_test.go]: Follow up PR: https://github.com/elastic/beats/pull/15993.

**Comment by ycombinator** [comment]
jenkins, test this

**Pull request merged**